File: app/src/main/java/com/odd/ostrinov2/tools/YoutubeShare.java
package com.odd.ostrinov2.tools;

import android.app.Activity;
import android.os.AsyncTask;
import android.widget.Toast;

import com.odd.ostrinov2.Constants;

import org.json.JSONException;
import org.json.JSONObject;

public class YoutubeShare {

    private String youtube, url;
    private Activity activity;
    private DBHandler db;

    public YoutubeShare(Activity activity, String url, DBHandler db){
        this.db = db;
        this.activity = activity;
        this.url = url;
        YoutubeGetInfo youtubeGetInfo = new YoutubeGetInfo();
        youtubeGetInfo.execute();
    }
    public String getTitle(){
        return youtube;
    }

    private class YoutubeGetInfo extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected Void doInBackground(final Void... arg0) {
            HttpHandler sh = new HttpHandler();
            String jsonUrl = "https://www.youtube.com/oembed?format=json&amp;url=" + url
                    + "&key=" + Constants.YDATA_API_TOKEN;

            // Making a request to url and getting response
            String jsonStr = sh.makeServiceCall(jsonUrl);

            //  Log.e(TAG, "Response from url: " + jsonStr);

            if (jsonStr != null) {
                try {
                    JSONObject jsonObj = new JSONObject(jsonStr);
                    System.out.println(jsonObj);
                    youtube = jsonObj.get("title").toString();
                    UtilMeths.INSTANCE.parseAddOst(youtube, db, url);

                } catch (final JSONException e) {
                    // Log.e(TAG, "Json parsing error: " + e.getMessage());
                    activity.runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            Toast.makeText(activity,
                                    "Json parsing error: " + e.getMessage(),
                                    Toast.LENGTH_LONG)
                                    .show();
                        }
                    });

                }
            } else {
                //Log.e(TAG, "Couldn't get json from server.");
                activity.runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(activity,
                                "Couldn't get json from server. Check LogCat for possible errors!",
                                Toast.LENGTH_LONG)
                                .show();
                    }
                });

            }

            return null;
        }

        @Override
        protected void onPostExecute(Void result) {
            super.onPostExecute(result);
        }
    }
}

File: app/src/main/java/com/odd/ostrinov2/asynctasks/YoutubeShare.kt
package com.odd.ostrinov2.tools

import android.content.Context
import android.os.AsyncTask
import android.util.Log
import android.widget.Toast
import com.odd.ostrinov2.Constants
import org.json.JSONException
import org.json.JSONObject
import java.lang.ref.WeakReference

class YoutubeShare(private val url: String) : AsyncTask<Void, Void, Void>() {

    private var title: String? = null
    private var parsedTitle: String? = null
    private lateinit var wContext: WeakReference<Context>

    fun setContext( context: Context){
        wContext = WeakReference(context)
    }

    override fun doInBackground(vararg params:Void?): Void? {
        val sh = HttpHandler()
        val jsonUrl = ("https://www.youtube.com/oembed?format=json&amp;url=" + url
                + "&key=" + Constants.YDATA_API_TOKEN)

        // Making a request to url and getting response
        val jsonStr = sh.makeServiceCall(jsonUrl)

        //  Log.e(TAG, "Response from url: " + jsonStr);

        if (jsonStr != null) {
            try {
                val jsonObj = JSONObject(jsonStr)
                Log.d("jsonObj", jsonObj.toString())
                title = jsonObj.get("title").toString()
                parsedTitle = UtilMeths.parseAddOst(title!!, url).title

            } catch (e: JSONException) {
                Log.e("JSONException", "Json parsing error: " + e.message)

            }

        } else {
            Log.e("JsonRetrieveError", "Couldn't get json from server.")
        }

        return null
    }

    override fun onPostExecute(result: Void?) {
        super.onPostExecute(result)
        Toast.makeText(wContext.get(), "Added $parsedTitle to your OST library", Toast.LENGTH_SHORT).show()
    }
}